### PR TITLE
Use correct ID types and improve tests for dms

### PR DESCRIPTION
**Pull request by erlendvollset**
- Don't use ID base classes directly
- Be more thorough when testing. Don't just assert that collecgtion sizes are the same

## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/data_modeling/ids.py [hunk 1]
```diff
@@ -1,15 +1,24 @@
...
+from typing import Any, ClassVar, Literal, Optional, Sequence, Tuple, Type, TypeVar, Union, cast
 
 from cognite.client.utils._auxiliary import rename_and_exclude_keys
 from cognite.client.utils._identifier import DataModelingIdentifier, DataModelingIdentifierSequence
 from cognite.client.utils._text import convert_all_keys_recursive, convert_all_keys_to_snake_case
 
 
 @dataclass(frozen=True)
-class DataModelingId:
+class AbstractDataclass(ABC):
```
Comment (doctrino) [inline cognite/client/data_classes/data_modeling/ids.py:13]: What is the advantage of a shared base class?

Comment (doctrino) [reply cognite/client/data_classes/data_modeling/ids.py:13]: Ensure frozen?

**Review by doctrino — APPROVED** [review]
Just a question. Nice to see the cleanup and improvement.

File: cognite/client/data_classes/data_modeling/ids.py (hunk 1 above)
Comment (erlendvollset) [reply cognite/client/data_classes/data_modeling/ids.py:13]: Ensuring that the direct subclasses of this class aren't instantiated directly. For Python to validate this at runtime without this, you need at least one abstract method defined (which we don't have here).

**Pull request merged**